src/service/topics.ts

```diff
@@ -127,6 +128,13 @@ const getLatestTopicPost = async (topicName: string): Promise<TopicLatestPostDat
       success: res?.data?.success
     };
   } catch (e) {
+    if (e?.response?.message === 'This topic has no active post')
+      return {
+        success: false,
+        message: e?.response?.message,
+        expired_at: moment().add(1, 'h').toISOString()
+      };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling logic seems fine, but it's better to use constants for error messages instead of hardcoding them. This will make it easier to manage and change these messages in the future.
```diff
- if (e?.response?.message === 'This topic has no active post')
+ if (e?.response?.message === ERROR_MESSAGES.NO_ACTIVE_POST)
  return {
    success: false,
    message: e?.response?.message,
-   expired_at: moment().add(1, 'h').toISOString()
+   expired_at: formatISO(addHours(new Date(), 1))
  };
```
In the above code, `ERROR_MESSAGES` is an object that contains all your error messages. You would need to define this somewhere in your codebase.


<!-- This is an auto-generated comment by OSS CodeRabbit -->